### PR TITLE
Fix typo made in #1964

### PR DESCRIPTION
**Pull request by vnlitvinov**
#1964 changed the API for `OmniSciDBClient` but didn't change all usages of `device=` parameter when constructing a client. In my case a `connection.database()` call was failing.

### CODE REVIEW

**Comment by xmnlab** [comment]
@vnlitvinov thanks for caching that up.

could you add tests for that to `omniscidb/tests/`?

would be good to have a test for the 2 cases, when the database name is the same of the current_database and for a different name (maybe you can use the `omnisci` database or you will need to create a temporary database for the test.

thanks!

**Comment by vnlitvinov** [comment]
@abykovsk can you take a stab at making required tests?

**Comment by datapythonista** [comment]
> @abykovsk could you please take a stab at making required tests?

There should be a test that failed before merging #1964, so this never broke. There is [this tests](https://github.com/datapythonista/ibis/blob/master/ibis/omniscidb/tests/test_client.py#L65) but since `name == None` the breaking code is never called. I guess you can add a pytest parametrization to that test, so it's called with `name == None` but also to a different database name to the current want.

**Comment by vnlitvinov** [comment]
> would be good to have a test for the 2 cases, when the database name is the same of the current_database and for a different name

I have added a separate test (the case where database name is same as current behaves exactly the same as with database name being `None` and is already handled by `test_database_layer`).
I tried parametrizing it first, but felt it would end up too complex (and it would actually test two unrelated things which isn't good for a unit test), so I wrote a new test instead.

**Comment by vnlitvinov** [comment]
Anything left for me to do to make this accepted?

File: ibis/omniscidb/tests/test_client.py [hunk 1]
```diff
@@ -73,6 +73,16 @@ def test_database_layer(con, alltypes):
     assert db.list_tables() == con.list_tables()
 
 
+def test_separate_database(con, test_data_db, temp_database):
```
Comment (jreback) [inline ibis/omniscidb/tests/test_client.py:76]: how does this test the above?

this actually is a fine general test, can you move to the ibis/tests/all/test_client.py

Comment (vnlitvinov) [reply ibis/omniscidb/tests/test_client.py:76]: _Hmm, I thought I've answered, but it seems I didn't press the `Comment` button or something..._

I've added the comments explaining the test flow in the test, but I can re-iterate here:
1. The connection keeps an object of currently active database, and if you request a new database object with either `None` as name or with the name that equals current, it returns the existing object, but if you request other name it creates a new object.
2. This test goes as following - it requests a temporary database (this creates new connection with a newly made temp database as current), then it requests an existing database (which obviously isn't equal to current db name being a temp one), so the connection creates a new object
3. Assertions there aren't strictly needed for testing this particular fix, but a test without assertions looks strange :) Actually, if you run this test prior to my fix there would be `AttributeError` raised, so it's still testing stuff, but I decided to check a few things just to use those objects.

I can move it to the general test suite, sure.

**Comment by jreback** [comment]
looks ok, but tests are failing

**Comment by vnlitvinov** [comment]
I know, we're looking at it. The test should probably be enabled only for SQL-like backends.

File: ibis/omniscidb/tests/test_client.py (hunk 1 above)
Comment (vnlitvinov) [reply ibis/omniscidb/tests/test_client.py]: @jreback please take a look, we have moved the test and enabled it only for backends it has a meaning for.

**Comment by vnlitvinov** [comment]
@jreback ping :)

**Comment by jreback** [comment]
@vnlitvinov omnisci is currently not being tested, so all merging on hold

**Comment by vnlitvinov** [comment]
What's the reason for this? Can we help with that being fixed?

**Comment by jreback** [comment]
https://github.com/ibis-project/ibis/issues/2201

**Comment by datapythonista** [comment]
@vnlitvinov can you merge master, and check if the CI is still green please? CI is working for omnisci, we can move forward with this now

**Comment by vnlitvinov** [comment]
@datapythonista I've rebased instead for a cleaner history.

**Comment by vnlitvinov** [comment]
The CI failures look strange and unrelated to our changes... can I do something to get this merged?

**Comment by jreback** [comment]
@vnlitvinov we need a test that fails on master and is fixed by this change

**Comment by vnlitvinov** [comment]
@jreback it would be nice to see this requirement a bit earlier than in almost 3 months... in a PR fixing a typo (original diff was **one** line).

Am I reading this right that you want me to make a separate PR (with a failing test) which would turn master CI red, and only _then_ fix that test? All for 1 line (and two words) worth of a change?..

**Comment by jreback** [comment]
i commented quite a while ago on this 

one line or a million it doesn’t matter

if something is claimed to be fixed then it needs a test

often the test are way longer than the patch 

i want a test included with this PR
one that fails in master

**Comment by vnlitvinov** [comment]
> i want a test included with this PR
> one that fails in master

https://github.com/ibis-project/ibis/pull/2106/files#diff-2268971ed7d3b780e597e36b7a466045R203

File: ibis/tests/all/conftest.py [hunk 2]
```diff
@@ -414,3 +414,42 @@ def temp_view(con) -> str:
...
+            '{} backend doesn\'t have current_database method.'.format(
+                backend.name
+            )
+        )
+    return con.current_database
+
+
+@pytest.fixture
+def temp_database(con, backend, current_data_db: str) -> str:
+    """Create a temporary database.
```
Comment (jreback) [inline ibis/tests/all/conftest.py:433]: please update the doc-string to be more indicative of what this actually does

File: ibis/tests/all/conftest.py [hunk 3]
```diff
@@ -414,3 +414,42 @@ def temp_view(con) -> str:
...
         if hasattr(con, 'drop_view'):
             con.drop_view(name, force=True)
+
+
+@pytest.fixture(scope='session')
+def current_data_db(con, backend) -> str:
+    """Return current database name."""
+    if not hasattr(con, 'current_database'):
+        pytest.skip(
+            '{} backend doesn\'t have current_database method.'.format(
```
Comment (jreback) [inline ibis/tests/all/conftest.py:424]: you can use f-strings

**Comment by jreback** [comment]
changes look ok. rebase on master and some comments, ping on green.

**Comment by vnlitvinov** [comment]
@jreback I think we addressed all your comments (so I marked them as resolved), and after merge with `master` CI is finally green.

**Comment by jreback** [comment]
thanks @vnlitvinov
